object_store_factory/src/aws.rs

```diff
@@ -141,7 +142,8 @@ impl S3Config {
         let mut builder = AmazonS3Builder::new()
             .with_region(self.region.clone().unwrap_or_default())
             .with_bucket_name(self.bucket.clone())
-            .with_allow_http(self.allow_http);
+            .with_allow_http(self.allow_http)
+            .with_conditional_put(S3ConditionalPut::ETagMatch);
```

Comment:
This was needed since otherwise we hit this error in object_store: https://github.com/apache/arrow-rs/blob/855666d9e9283c1ef11648762fe92c7c188b68f1/object_store/src/aws/mod.rs#L171

We hit these since with the delta-rs bump we picked up https://github.com/delta-io/delta-rs/pull/2813, and there the put mode is explicitly being set to `PutMode::Create` (whereas previously it defaulted to `PutMode::Overwrite`).

Reply:
If we always need to add this when creating a S3 object storage, then I think you will also need to add it to the `add_amazon_s3_specific_options()` function. Because object storage can also be created using this factory with options.